camel/societies/role_playing.py

```diff
@@ -194,50 +194,51 @@ def __init__(
         else:
             self.critic = None
 
-    def init_chat(self) -> Tuple[AssistantChatMessage, List[ChatMessage]]:
+    def init_chat(self) -> Tuple[BaseMessage, List[BaseMessage]]:
         r"""Initializes the chat by resetting both of the assistant and user
         agents, and sending the system messages again to the agents using
         chat messages. Returns the assistant's introductory message and the
         user's response messages.
 
         Returns:
-            A tuple containing an `AssistantChatMessage` representing the
-            assistant's introductory message, and a list of `ChatMessage`s
+            A tuple containing an `BaseMessage` representing the
+            assistant's introductory message, and a list of `BaseMessage`s
             representing the user's response messages.
         """
         self.assistant_agent.reset()
         self.user_agent.reset()
 
         # Send the system messages again to the agents using chat messages
-        assistant_msg = AssistantChatMessage(
-            role_name=self.assistant_sys_msg.role_name, role="assistant",
+        assistant_msg = BaseMessage.make_assistant_message(
+            role_name=self.assistant_sys_msg.role_name,
             content=(f"{self.user_sys_msg.content}. "
                      "Now start to give me instructions one by one. "
                      "Only reply with Instruction and Input."))
 
-        user_msg = UserChatMessage(role_name=self.user_sys_msg.role_name,
-                                   content=f"{self.assistant_sys_msg.content}")
+        user_msg = BaseMessage.make_user_message(
+            role_name=self.user_sys_msg.role_name,
+            content=f"{self.assistant_sys_msg.content}")
         assistant_response = self.assistant_agent.step(user_msg)
         if assistant_response.terminated or assistant_response.msgs is None:
             raise ValueError(f"Assistant agent terminated unexpectedly. "
                              f"Error info: {assistant_response.info}")
 
         return assistant_msg, assistant_response.msgs
 
-    def process_messages(
+    def reduce_message_options(
```

Comment:
```suggestion
    def reduce_messages(
```
I would prefer `reduce_messages`

Reply:
What about my comment that reduce_messages may be understood as making messages shorter?

Reply:
I see. Sure